python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -265,6 +266,8 @@ def lit(col: Any) -> Column:
                 error_class="COLUMN_IN_LIST", message_parameters={"func_name": "lit"}
             )
         return array(*[lit(c) for c in col])
+    elif isinstance(col, dict):
+        return create_map(*[lit(x) for x in chain(*col.items())])
```

Comment:
the problem is that if we want this as a struct or map

Reply:
Can you compare with createDataFrame? We also have a configuration `spark.sql.pyspark.inferNestedDictAsStruct.enabled`. We might need a configuration to control it.

Reply:
it seems this is not supported in PySpark Classic.

Reply:
another problem is the result of `create_map` is a `CreateMap` expression, not a literal

Reply:
> it seems this is not supported in PySpark Classic.

Sorry, I'm not sure what you meant here, pyspark "Classic" means local pyspark (as opposed to connect)?
I added the code change to both the classic functions and the connect ones.

> another problem is the result of `create_map` is a `CreateMap` expression, not a literal

Is that a problem? the `array` function that is used above in the same function to create an array "literal" also returns a `CreateArray` expression and not an array literal

Reply:
Should we also check if any of the dictionary values contain a Column type object and throw similar exceptions as in the array case above?

Reply:
> Should we also check if any of the dictionary values contain a Column type object and throw similar exceptions as in the array case above?

Sure, I'll add that check + the appropriate error

Reply:
So, I think we can merge the changes with master.